AirCasting/Utils/Averaging.swift

```diff
@@ -0,0 +1,151 @@
+                    }
+                    
+                    print("[🦆] Unaveraged measurements: \(unaveragedMeasurements.count)")
+                    unaveragedMeasurements.chunks(ofCount: elementsInChunk.rawValue).forEach( { measuremensInChunk in
+                        print("[🦆] B")
+                        if (measuremensInChunk.count == elementsInChunk.rawValue) {
+                            /// https://github.com/apple/swift-algorithms/blob/main/Guides/Chunked.md
+                            /// For integer types, any remainder of the division is discarded so we need to add 1 to elementsInChunk/2 to get the middle value.
+                            let idx = measuremensInChunk.index(measuremensInChunk.startIndex, offsetBy: (elementsInChunk.rawValue/2 + 1))
+                            let middleMeasurement = measuremensInChunk[idx]
+                            let averageValue = measuremensInChunk.map({ $0.value}).reduce(0.0, +) / Double(measuremensInChunk.count)
+                            middleMeasurement.value = averageValue
+                            middleMeasurement.averagingWindow = elementsInChunk.rawValue
+                            print("[🦆] middleMeasurement: \(middleMeasurement)")
+                            averagedMeasurements.append(middleMeasurement)
+                            print("[🦆] averagedMeasurements: \(averagedMeasurements)")
+                        }
+                    })
+                    do {
+                        print("[🦆] AVERAGING: update stream: \(averagedMeasurements.count)")
+                        if !averagedMeasurements.isEmpty {
+                            try storage.updateMeasurements(stream: stream, newMeasurements: NSOrderedSet(array: averagedMeasurements))
+                        }
+                    } catch {
+                        print("[🦆] ERRRRROR")
+                    }
+                }
+            }
+        }
+    }
+    
+    private func schduleAveraging(session: SessionEntity) {
+        let uuid = session.uuid!
+        
+        let timer = Timer.publish(every: TimeInterval(TimeThreshold.firstThreshold.rawValue), on: .main, in: .common)
+            .autoconnect()
+            .first()
+            .sink { [weak self] _ in
+                self?.measurementStreamStorage.accessStorage { storage in
+                    guard let session = try? storage.getExistingSessionWith(uuid) else { return }
+                    self?.startPeriodicAveraging(session: session)
+                }
+            }
+        timers[uuid] = timer
+    }
+    
+    private func startPeriodicAveraging(session: SessionEntity) {
+        guard let context = session.managedObjectContext,
+              let uuid = session.uuid else { return }
+
+        let timer = Timer.publish(every: 5.0, on: .main, in: .common)
+            .autoconnect()
+            .sink { [weak self] _ in
+                context.perform {
```

Comment:
consider scenario when after 2h we have so many measurements that it will take more than 5s to perform it